pokemon-checker/src/DataTransferObjects/PokemonDTO.ts

```diff
@@ -116,9 +122,29 @@ class PokemonDTO {
       type2: this.type2,
       sprites: {
         frontDefault: this.frontDefault,
-        frontShiny: this.frontShiny
-      }
-    }
+        frontShiny: this.frontShiny,
+      },
+    };
+  }
+
+  /**
+   * Determines whether this PokemonDTO is suitable for rendering
+   */
+  public get hasFullData(): boolean {
```

Comment:
Classes operating on DTOs need to know if the full thing is there or not. If we're dealing with incomplete data, we'll need to fetch using the `url`.